tmt/result.py

```diff
@@ -113,26 +123,11 @@ def __init__(
                 self.ids[key] = test.node.get(key)
             interpret = ResultInterpret(test.result) if test.result else ResultInterpret.RESPECT
         else:
-            try:
-                self.ids = data['ids']
-            except KeyError:
-                self.ids = {}
+            self.ids = data.ids
             interpret = ResultInterpret.RESPECT
 
-        # Check for valid results
-        try:
-            self.result = ResultOutcome(data['result'])
-        except KeyError:
-            raise tmt.utils.SpecificationError("Missing test result.")
-        except ValueError:
-            raise tmt.utils.SpecificationError(
-                f"Invalid result '{data['result']}'.")
```

Comment:
One thing to remember: this code should probably not disappear completely. Yes, it should not appear at this place anymore, since `data.result` should be of the right type already, but something similar to this validation should happen at places where results are ingested from files. Please, make sure that the relevant code can handle invalid result outcomes, for example.